test/base/product.jl

```diff
@@ -137,7 +137,335 @@ end
     end
 end
 
-@testset "value iteration" begin
+@testset "bellman deterministic labelling wtih strategy" begin
+    for N in [Float32, Float64, Rational{BigInt}]
+        @testset "N = $N" begin
+            prob1 = IntervalAmbiguitySets(;
+                lower = N[
+                    0//10 4//10
+                    1//10 3//10
+                    2//10 2//10
+                ],
+                upper = N[
+                    5//10 9//10
+                    6//10 8//10
+                    7//10 7//10
+                ],
+            )
+
+            prob2 = IntervalAmbiguitySets(;
+                lower = N[
+                    5//10 2//10
+                    3//10 3//10
+                    1//10 4//10
+                ],
+                upper = N[
+                    7//10 3//10
+                    5//10 6//10
+                    3//10 9//10
+                ],
+            )
+
+            prob3 = IntervalAmbiguitySets(; lower = N[
+                0 0
+                0 0
+                1 1
+            ], upper = N[
+                0 0
+                0 0
+                1 1
+            ])
+
+            transition_probs = [prob1, prob2, prob3]
+            mdp = IntervalMarkovDecisionProcess(transition_probs)
+
+            # Product model - just simple reachability
+            delta = TransitionFunction(Int32[
+                1 2
+                2 2
+            ])
+            istate = Int32(1)
+            atomic_props = ["reach"]
+            dfa = DFA(delta, istate, atomic_props)
+
+            labelling = DeterministicLabelling(Int32[1, 1, 2])
+
+            prod_proc = ProductProcess(mdp, dfa, labelling)
+
+            V = N[
+                4 1
+                2 3
+                0 5
+            ]
+
+            workspace = IntervalMDP.construct_workspace(prod_proc)
+
+            eps = one(N) / N(1000)
+
+            # No Strategy
+            Vtar = IntervalMDP.bellman(V, prod_proc; upper_bound = false)
+
+            # Non Stationary Strategy
+            prop = FiniteTimeDFAReachability([2], 10)
+            spec = Specification(prop, Pessimistic, Maximize)
+            problem = ControlSynthesisProblem(prod_proc, spec)
+            strategy_cache = IntervalMDP.construct_strategy_cache(problem)
+
+            Vres = copy(V)
+
+            Vres = IntervalMDP.bellman!(
+                workspace,
+                strategy_cache,
+                Vres,
+                V,
+                prod_proc,
+                upper_bound = false,
+            )
+
+            @test Vtar ≈ Vres atol=eps
+
+            # Stationary Strategy
+            prop = InfiniteTimeDFAReachability([2], eps)
+            spec = Specification(prop, Pessimistic, Maximize)
+            problem = ControlSynthesisProblem(prod_proc, spec)
+            strategy_cache = IntervalMDP.construct_strategy_cache(problem)
```

Comment:
Same goes here.
```suggestion
            strategy_cache = IntervalMDP.StationaryStrategyCache(zeros(Int32, 3, 2))
```

And you could add two more cases, one where the current action corresponds to the optimal and one where it doesn't (rather than one for the first time step where there is no "optimal action" yet).

Reply:
new test cases added in https://github.com/Zinoex/IntervalMDP.jl/pull/89/commits/3617936d65ad9ed76a72a7cec3a0faf776dee3b9